NorthstarDLL/httprequesthandler.cpp

```diff
@@ -0,0 +1,602 @@
+#include "pch.h"
+#include "httprequesthandler.h"
+#include "version.h"
+#include "squirrel.h"
+#include "tier0.h"
+
+HttpRequestHandler* g_httpRequestHandler;
+
+bool IsHttpDisabled()
+{
+	const static bool bIsHttpDisabled = Tier0::CommandLine()->FindParm("-disablehttprequests");
+	return bIsHttpDisabled;
+}
+
+bool IsLocalHttpAllowed()
+{
+	const static bool bIsLocalHttpAllowed = Tier0::CommandLine()->FindParm("-allowlocalhttp");
+	return bIsLocalHttpAllowed;
+}
+
+bool DisableHttpSsl()
+{
+	const static bool bDisableHttpSsl = Tier0::CommandLine()->FindParm("-disablehttpssl");
+	return bDisableHttpSsl;
+}
+
+HttpRequestHandler::HttpRequestHandler()
+{
+	// Cache the launch parameters as early as possible in order to avoid possible exploits that change them at runtime.
+	IsHttpDisabled();
+	IsLocalHttpAllowed();
+	DisableHttpSsl();
```

Comment:
Realistically this isn't needed but i'll allow it since i'm gonna centralize config soon anyway. You can resolve this comment.